### PR TITLE
fix(vertexai): use opts in NewClient

### LINKED ISSUE #9160: genai: func NewClient does not use opts ...option.ClientOption param
**Code**
https://github.com/googleapis/google-cloud-go/blob/main/vertexai/genai/client.go#L47

```
func NewClient(ctx context.Context, projectID, location string, opts ...option.ClientOption) (*Client, error) {
	apiEndpoint := fmt.Sprintf("%s-aiplatform.googleapis.com:443", location)
	c, err := aiplatform.NewPredictionClient(ctx, option.WithEndpoint(apiEndpoint))
	if err != nil {
		return nil, err
	}
	return &Client{
		c:         c,
		projectID: projectID,
		location:  location,
	}, nil
}
```

**Implication**
The following code does not generate valid authentication:
```
client, err := genai.NewClient(ctx, "project_id", "region", option.WithCredentialsFile("../cred.json"))
```

**Solution**
Use the `opts` param
```
func NewClient(ctx context.Context, projectID, location string, opts ...option.ClientOption) (*Client, error) {
	apiEndpoint := fmt.Sprintf("%s-aiplatform.googleapis.com:443", location)
	opts = append(opts, option.WithEndpoint(apiEndpoint))
	c, err := aiplatform.NewPredictionClient(ctx, opts...)
	if err != nil {
		return nil, err
	}
	return &Client{
		c:         c,
		projectID: projectID,
		location:  location,
	}, nil
}
```

### PR DESCRIPTION
**Pull request by eliben**
Fixes #9160

### CODE REVIEW

**Comment by eliben** [comment]
Closing in favor of #9163 that was earlier

File: vertexai/genai/client.go [hunk 1]
```diff
@@ -47,7 +47,8 @@ type Client struct {
 // package.
 func NewClient(ctx context.Context, projectID, location string, opts ...option.ClientOption) (*Client, error) {
 	apiEndpoint := fmt.Sprintf("%s-aiplatform.googleapis.com:443", location)
-	c, err := aiplatform.NewPredictionClient(ctx, option.WithEndpoint(apiEndpoint))
+	opts = append(opts, option.WithEndpoint(apiEndpoint))
```
Comment (jba) [inline vertexai/genai/client.go:50]: Appending to opts could affect the caller.
Also we want the user opts to override whatever we do.
Follow this pattern: https://github.com/googleapis/google-cloud-go/blob/main/bigquery/storage_client.go#L59.

**Comment by jba** [comment]
Thanks for doing this, sorry I missed it.

You can test by passing in your own endpoint option that's wrong, and looking for an error.

File: vertexai/genai/client.go (hunk 1 above)
Comment (eliben) [reply vertexai/genai/client.go:50]: Thanks! can you add these comments on #9163 ?

Comment (jba) [reply vertexai/genai/client.go:50]: Not necessary, #9163 is correct.

**Pull request closed without merging**